sound/soc/sof/pcm.c

```diff
@@ -646,6 +646,18 @@ static int sof_pcm_ack(struct snd_soc_component *component,
 	return snd_sof_pcm_platform_ack(sdev, substream);
 }
 
+static snd_pcm_sframes_t sof_pcm_delay(struct snd_soc_component *component,
+				       struct snd_pcm_substream *substream)
+{
+	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(component);
+	const struct sof_ipc_pcm_ops *pcm_ops = sdev->ipc->ops->pcm;
+
+	if (pcm_ops->delay)
```

Comment:
and check for !pcm_ops

Reply:
sure. We may need other PR to fix in pcm

pcm.c (sound\soc\sof) line 146 : 	if (pcm_ops->hw_free && spcm->prepared[substream->stream]) {
pcm.c (sound\soc\sof) line 180 : 	if (pcm_ops->hw_params) {
pcm.c (sound\soc\sof) line 215 : 	if (pcm_ops->hw_free && spcm->prepared[substream->stream]) {
pcm.c (sound\soc\sof) line 345 : 	if (pcm_ops->trigger)
pcm.c (sound\soc\sof) line 590 : 	if (pcm_ops->dai_link_fixup)

Reply:
o, my code is out of date. we don't need fixup